code/script.js

```diff
@@ -1,325 +1,356 @@
+      } else {
+        alert(`No, the person is not a ${value}! Keep all people that is a ${value}!`)
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+      }
+    }
+    //Redraw the game board.
+    generateBoard()
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+    //When clicking guess, the player has to confirm that they want to make a guess.
+    const guess = (personToConfirm) => {
+    let confirmPerson = confirm(`Do you really want to make a guess on ${personToConfirm}?`);
+    if (confirmPerson === true) {
+      checkMyGuess (personToConfirm)
+    } else {
+      alert('Ok, no problem! Keep on guessing!')
+    }
+  }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  // If player confirms, this function is invoked
+  const checkMyGuess = (personToCheck) => {
+    board.style.display = 'none'
+    winOrLose.style.display = 'flex'
 
-// If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+    if (personToCheck === secret.name) {
+      winOrLoseText.innerHTML = `
+        Congratulations, You won! ${secret.name} is the person we were looking for. `
+    } else {
+      winOrLoseText.innerHTML = `
+        Unfortunately, that is the wrong answer. It is ${secret.name} 
+        that we were looking for. Please try again!`
+    }    
+  };
 
-// Invokes the start function when website is loaded
-start()
+  // Invokes the start function when website is loaded
+  start()
 
-// All the event listeners
-restartButton.addEventListener('click', start)
+  // All the event listeners
+  restartButton.addEventListener('click', start);
+  findOutButton.addEventListener('click', checkQuestion)
+  questions.addEventListener('change', selectQuestion)
+  playAgainButton.addEventListener('click', () => {
+  winOrLose.style.display = 'none';
+  board.style.display = 'flex'
+  location.reload();
```

Comment:
I'm not sure these are eventlisteners, so maybe they can be moved to somewhere else, if it's possible?